tests/inference/qnn_test.py

```diff
@@ -441,22 +452,32 @@ def test_expectation_bitstring_energy(self):
         n_moments=n_moments,
         p=act_fraction)
     hamiltonian_raw_circuit = hamiltonian_raw_circuits[0]
-    hamiltonian_circuit = models.DirectQuantumCircuit(hamiltonian_raw_circuit)
+    hamiltonian_circuit_initializer = tf.keras.initializers.RandomUniform(
+        minval=qnn_minval, maxval=qnn_maxval, seed=self.tf_random_seed_alt)
+    hamiltonian_circuit = models.DirectQuantumCircuit(
+        hamiltonian_raw_circuit, hamiltonian_circuit_initializer)
+    hamiltonian_circuit.build([])
 
     # Total circuit
     bitstring_circuit = models.circuit_utils.bit_circuit(qubits)
     measurement_circuit = cirq.Circuit(
         cirq.measure(q, key=f"measure_qubit_{n}") for n, q in enumerate(qubits))
+    # needed later for retrieving sample results from Cirq output
+    qb_keys = [(q, f"measure_qubit_{i}") for i, q in enumerate(qubits)]
```

Comment:
`qb_keys` doesn't appear to be used?

Reply:
Used below but accidentally created the same `qb_keys` twice, now removed the second copy.  I decided to put `qb_keys` creation here rather than where it is used because `qb_keys` needs to use the same measurement keys as the `measurement_circuit` above it.  Added a comment describing this.

Reply:
the usual heuristic is to move variable construction to the "latest possible line" so that it's used shortly after it's created. In this case, it appears you can move both `qb_keys` and `measurement_circuit` much later

Reply:
If we move `measurement_circuit` later, then we also need to move `total_circuit` later.  But then `state_raw_circuit` and `hamiltonian_raw_circuit` are used farther away from where they were created.  The grouping of having `total_circuit` defined here was to keep all quantum circuit set up more local

Reply:
I see - I would still move `qb_keys` to the bottom since it's absent of these dependencies and strictly makes usages more local. Is it clear why this heuristic is more readable?

Reply:
Yes, the localization heuristic makes things more readable in general.  Depending on the situation though, I think deviations from this heuristic can be more readable - like here.  But moving just `qb_keys` later, rather than both `qb_keys` and `measurement_circuit`, is simple enough - done.

Reply:
Thank you! In the future, it's completely reasonable to stick to your personal preference. This was just mine, independent of the heuristic.